src/main/java/com/box/sdk/BoxWebHookSignatureVerifier.java

```diff
@@ -0,0 +1,255 @@
+package com.box.sdk;
+
+import java.nio.charset.Charset;
+import java.security.InvalidKeyException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+
+import com.box.sdk.internal.pool.MacPool;
+
+/**
+ * Signature verifier for Webhook Payload.
+ *
+ * @author Vladimir Hrusovsky
+ * @author Stanislav Dvorscak
+ */
+public class BoxWebHookSignatureVerifier {
+
+    /**
+     * Reference to UTF_8 {@link Charset}.
+     */
+    private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+    /**
+     * Versions supported by this implementation.
+     */
+    private static final Set<String> SUPPORTED_VERSIONS = Collections.singleton("1");
+
+    /**
+     * Algorithms supported by this implementation.
+     */
+    private static final Set<BoxSignatureAlgorithm> SUPPORTED_ALGORITHMS = Collections.unmodifiableSet(
+            EnumSet.of(BoxSignatureAlgorithm.HMAC_SHA256));
+
+    /**
+     * {@link Mac}-s pool.
+     */
+    private static final MacPool MAC_POOL = new MacPool();
+
+    /**
+     * Primary key setup within the Box.
+     */
+    private final String primarySignatureKey;
+
+    /**
+     * Secondary key setup within the Box.
+     */
+    private final String secondarySignatureKey;
+
+    /**
+     * Creates a new instance of verifier specified with given primary and secondary keys. Primary key and secondary key are needed for
+     * rotating purposes, at least at one has to be valid.
+     *
+     * @param primarySignatureKey
+     *            primary signature key for web-hooks (can not be null)
+     * @param secondarySignatureKey
+     *            secondary signature key for web-hooks (can be null)
+     * @throws IllegalArgumentException
+     *             primary key can not be null
+     */
+    public BoxWebHookSignatureVerifier(String primarySignatureKey, String secondarySignatureKey) {
+        if (primarySignatureKey == null) {
+            throw new IllegalArgumentException("At least primary signature key must be provided!");
```

Comment:
I think we should check that either primarySignatureKey or secondarySignatureKey must be present.
There might be scenario where primarySignatureKey is not configured but secondary is configured.